[Github][OpenMP] Adding rule for OpenMP label

jplehr opened the pull request:
The intention is to add the paths: top-level OpenMP and MLIR/Dialect/OpenMP
I'm sure there are other paths missing here.

jplehr committed:
[Github][OpenMP] Adding rule for OpenMP label

Adds a first rule to add the OpenMP label for new PRs.

shiltian commented on .github/new-prs-labeler.yml:
```diff
@@ -243,6 +243,10 @@ tools:llvm-mca:
+openmp:
+  - mlir/**/OpenMP/*
+  - openmp/**
+
```
changes in clang will also be needed right?

shiltian reviewed (commented):
I'm thinking we will probably need clang-openmp as well, similar to flang-openmp.

jplehr commented:
Is the overall feeling that we should have separate labels `clang-openmp`, `flang-openmp` and `openmp`?
The latter would cover everything inside the `openmp` folder, i.e., the runtime, whereas the other two are for the "frontend" parts of the project?

jhuber6 commented:
Maybe we should split out a separate `openmp-libomptarget` and `openmp-libomp`?

shiltian commented:
> Or was the idea to have the clang/flang parts as part of *-libomp and *-libomptarget?

No, that will be too much. I'm not even sure if we need `openmp-libomptarget` TBH.

jhuber6 commented:
> > Or was the idea to have the clang/flang parts as part of *-libomp and *-libomptarget?
> 
> No, that will be too much. I'm not even sure if we need `openmp-libomptarget` TBH.

Yeah, I don't know how much granularity people expect with this stuff, since we'd probably need to do one for libompd and all the other stuff. Just thinking because `libomptarget` is very separate from libomp at this point. I'm fine either way with just OpenMP.

jplehr commented:
I have added a whole bunch of files and paths (mainly copying from the CODEOWNERS file)

shiltian reviewed (approved):
The file list looks good to me. We can discuss granularity in the weekly meeting.

shiltian commented:
Based on the latest Discourse discussion, this needs to be approved by the real code owner, which is @jdoerfert .

cor3ntin commented on .github/new-prs-labeler.yml:
```diff
@@ -243,6 +243,50 @@ tools:llvm-mca:
+  - mlir/test/Target/LLVMIR/openmp*
+
+openmp:libomp:
+  - any: ['openmp/**', '!openmp/libomptarget/*']
```
Shouldn't that be `all`? ie, that might always be true

jplehr replied:
I tested this in https://github.com/jplehr/llvm-project/pulls and it appeared to be working as expected.

jplehr replied:
I carefully re-red the documentation, and I believe that the difference is that the `any` rule applies to any path modified in the PR, whereas the `all` rule applies to all paths in the PR.
In this case, we want to label the change with OpenMP as soon as it touches something in these paths, even though it may also touch other paths.

jplehr commented:
I think that the Flang team already added the labeling parts for their end, is that right? @llvm/pr-subscribers-flang-openmp

jdoerfert reviewed (approved):
LG, let's move this forward

jplehr commented:
- rebased
- removed the `flang:openmp` parts, since they were added before.
- changed `openmp:clang` to `clang:openmp` to be consistent with other entries in the file.

jplehr commented:
I have updated this once more with
- updated patterns for paths to follow the `foo/**` syntax as it seems to have been preferred in this file,
- a glob pattern in the two runtime labels. I believe that this was necessary to cover all paths under the respective top-level directory, e.g., `openmp/libomptarget/**`.

I'm going to create the new labels and merge this later today. Please feel free to comment.
